source/game/system/map/MapdataCheckPath.cc

```diff
@@ -21,10 +21,75 @@ void MapdataCheckPath::read(EGG::Stream &stream) {
     }
 }
 
+/// @brief Performs DFS to calculate @ref m_depth for all subsequent checkpaths.
+/// @param depth Number of checkpaths from first checkpath
+/// @addr{0x805150E0}
+void MapdataCheckPath::findDepth(s8 depth, const MapdataCheckPathAccessor &accessor) {
+    if (m_depth != -1) {
+        return;
+    }
+    m_depth = depth;
+    for (size_t i = 0; i < 6; i++) {
+        u16 nextID = m_next[i];
+        if (nextID == 0xff) {
+            continue;
+        }
+        MapdataCheckPath *next = accessor.get(nextID);
+        next->findDepth(depth + 1, accessor);
+    }
+}
+
+u8 MapdataCheckPath::start() const {
+    return m_start;
+}
+
+u8 MapdataCheckPath::end() const {
+    return m_start + m_size - 1;
+}
+
+u16 MapdataCheckPath::getNext(u16 i) const {
+    return m_next[i];
+}
+
+u16 MapdataCheckPath::getPrev(u16 i) const {
+    return m_prev[i];
+}
+
+u8 MapdataCheckPath::depth() const {
+    return m_depth;
+}
+
+bool MapdataCheckPath::isPointInPath(u16 checkpointId) const {
+    return m_start <= checkpointId && checkpointId <= end();
+}
+
+/// @addr{0x80515014}
+MapdataCheckPath *MapdataCheckPathAccessor::findCheckpathForCheckpoint(u16 checkpointId) {
+    MapdataCheckPath *checkpath;
+    for (size_t i = 0; i < size(); i++) {
+        checkpath = get(i);
+        if (checkpath->isPointInPath(checkpointId)) {
```

Comment:
Since you only over use this inside the for loop, it's fine to just declare and assign same line

```suggestion
    for (size_t i = 0; i < size(); i++) {
        MapdataCheckPath *checkpath = get(i);
```